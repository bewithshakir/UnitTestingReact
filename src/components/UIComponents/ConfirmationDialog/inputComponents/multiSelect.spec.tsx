import { shallow } from 'enzyme';
import MultiSelectComponent from './multiSelect';


const mockFormik = {
    touched: {},
    error: "",
    errors: {},
    initialError: "",
    initialTouched: false,
    initialValue: "",
    values: {},
    getFieldProps: jest.fn(),
    setFieldValue: jest.fn()
};

describe('renders inputText properly', () => {
    const onChange = jest.fn();
    const handleTouch = jest.fn();
    const inputComponent = shallow(
        <MultiSelectComponent field={{
            fieldType: 'multiSelect',
            name: 'city',
            label: 'Select City',
            options: [{ value: "a", label: 'a' }, { value: 'b', label: 'b' }]
        }}
            fieldId={`multiSelect`}
            formik={mockFormik as any}
            onChange={onChange}
            handleTouched={handleTouch}
        />);

    it('snapshot testing', () => {
        expect(inputComponent).toMatchSnapshot();
    });
});
