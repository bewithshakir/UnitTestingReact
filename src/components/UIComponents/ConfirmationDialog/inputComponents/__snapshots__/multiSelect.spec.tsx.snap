// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders inputText properly snapshot testing 1`] = `
<MultiSelect
  error={false}
  id="select-label"
  items={
    Array [
      Object {
        "label": "a",
        "value": "a",
      },
      Object {
        "label": "b",
        "value": "b",
      },
    ]
  }
  key="multiSelect"
  label="Select City"
  loadingMessage="Loading..."
  name="city"
  noOptionsMessage="No Options found"
  onBlur={[Function]}
  onChange={[MockFunction]}
  placeholder="Select"
  required={false}
/>
`;
